src/app/util/mock/attribute-storage.cpp

```diff
@@ -244,6 +254,26 @@ chip::EndpointId emberAfEndpointFromIndex(uint16_t index)
     return config.endpoints[index].id;
 }
 
+#if CHIP_CONFIG_USE_ENDPOINT_UNIQUE_ID
+CHIP_ERROR emberAfGetEndpointUniqueIdForEndPoint(EndpointId endpoint, MutableCharSpan & epUniqueIdSpan)
+{
+
+    // Retrieve the endpoint configuration from the mock node configuration
+    auto epConfig = GetMockNodeConfig().endpointById(endpoint);
+    VerifyOrReturnError(epConfig != nullptr, CHIP_ERROR_NOT_FOUND);
+
+    // Ensure the provided span is large enough to hold the unique ID
+    if (epUniqueIdSpan.size() < epConfig->endpointUniqueIdSize)
```

Comment:
If you use `CopyCharSpanToMutableCharSpan` you do not need to do these checks. This applies throughout: instead of size check + truncate copy (which is essentially the copy above) just use `return CopyCharSpanToMutableCharSpan`

Please make this change everywhere you use this patern.